### PR TITLE
fix: align linker output with Angular TypeScript compiler

### PR DESCRIPTION
**Pull request by Brooooooklyn**
- fix `deps: null` to use `ɵɵgetInheritedFactory` instead of no-args constructor
- fix `deps: 'invalid'` to emit `ɵɵinvalidFactory()`
- add ForPipe flag (16) for Pipe dependency inject calls
- fix `attribute: true` to use token as arg to `ɵɵinjectAttribute`
- handle `token: null` deps with `ɵɵinvalidFactoryDep(index)`
- fix `useClass != type` without deps to delegate to `useClass.ɵfac`
- omit `providedIn` when null (matching TS behavior)
- refactor `extract_deps_source` to take target param, generate correct
  inject function directly instead of fragile string replacement
- only emit `standalone: false` (omit when true, matching TS default)
- remove extra `type` property from `defineInjector` output
- omit `declarations`/`imports`/`exports` from `defineNgModule` in AOT mode
- add `id` support and `registerNgModuleType` IIFE side-effect for NgModules
- fix `link_injectable` to use conditional factory pattern for useValue,
  useExisting, useFactory (with deps), and useClass (with deps)
- fix `useFactory` without deps to wrap as `() => factory()`
- fix `useClass === type` without deps to include `__ngFactoryType__` param
- fix `link_class_metadata_async` to extract callback params and read
  decorators/ctorParameters/propDecorators from resolveMetadata return value

Close https://github.com/voidzero-dev/oxc-angular-compiler/issues/89

Co-Authored-By: Claude Opus 4.6 <noreply@anthropic.com>

---

> [!NOTE]
> **Medium Risk**
> Touches core linking logic for DI factories/injectables and NgModule/injector emission, which can change runtime behavior if mismatched. Risk is mitigated by extensive new regression tests covering the new edge cases.
> 
> **Overview**
> Updates `ɵɵngDeclareFactory` linking to match Angular semantics for `deps`: treats `deps: null` as inherited (uses `ɵɵgetInheritedFactory`), `deps: 'invalid'` as `ɵɵinvalidFactory()`, emits `ɵɵinvalidFactoryDep(index)` for `token: null`, fixes `@Attribute()` handling to pass the dep `token`, and generates the correct inject function/flags directly (including Pipe `ForPipe` flag `16` and `ɵɵdirectiveInject` vs `ɵɵinject`).
> 
> Reworks `ɵɵngDeclareInjectable` linking to omit `providedIn` when unset/null and to use Angular’s conditional-factory pattern for `useValue`, `useExisting`, and `useFactory`/`useClass` (including delegating `useClass != type` to `useClass.ɵfac` and wrapping `useFactory` without deps as `() => factory()`).
> 
> Adjusts emitted definitions to omit `standalone` when it would be `true` (only emit `standalone: false`), removes the extra `type` field from `defineInjector`, changes `defineNgModule` output to omit `declarations`/`imports`/`exports` in AOT mode while adding `id` support via `registerNgModuleType` IIFE, and fixes `ɵɵngDeclareClassMetadataAsync` linking to derive callback params and metadata from `resolveMetadata`’s returned object. Adds comprehensive tests for all of the above behaviors.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit d60a64159e7bfa67936d3c03b81e07669acf4276. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: crates/oxc_angular_compiler/src/linker/mod.rs [hunk 1]
```diff
@@ -808,23 +897,27 @@ fn link_ng_module(
 ) -> Option<String> {
     let mut parts = vec![format!("type: {type_name}")];
 
-    if let Some(declarations) = get_property_source(meta, "declarations", source) {
-        parts.push(format!("declarations: {declarations}"));
-    }
-    if let Some(imports) = get_property_source(meta, "imports", source) {
-        parts.push(format!("imports: {imports}"));
```
[the remark this answers is not in the document]
Comment (Brooooooklyn) [reply crates/oxc_angular_compiler/src/linker/mod.rs:815]: **INVALID.**
The injectable partial declaration format (R3InjectableMetadata.deps) is typed as R3DependencyMetadata[] | undefined — it never emits deps: null or deps: 'invalid'. Those values are exclusive to ɵɵngDeclareFactory declarations (which use R3FactoryMetadata.deps: R3DependencyMetadata[] | 'invalid' | null). The Angular TS linker for injectables also calls getArray('deps') unconditionally — confirming null/invalid are not possible.

**Comment by Brooooooklyn** [comment]
cursor review
